### PR TITLE
Add interfaces for Directive scope and Component bindings 

### PR DESCRIPTION
**Pull request by christophercr**
Interfaces for scope property in Directive and bindings property in Component to avoid using simple Object and the explicit type {[binding: string]: string} which is too verbose.

### CODE REVIEW

File: angularjs/angular.d.ts [hunk 1]
```diff
@@ -1738,6 +1738,10 @@ declare namespace angular {
          */
         require?: {[controller: string]: string};
     }
+    
+    interface IComponentBindings {
+		[binding: string]: string;
```
Comment (andy-ms) [inline angularjs/angular.d.ts]: Whitespace is inconsistent.

Comment (christophercr) [inline angularjs/angular.d.ts]: Hi, thanks for your review. I've fixed the issue, could you have a look?

Comment (andy-ms) [inline angularjs/angular.d.ts]: There still is a mix of tabs and spaces, as well as some trailing whitespace. The rest of the file uses 4 spaces for indentation. Try turning on whitespace rendering in your editor.

Comment (christophercr) [inline angularjs/angular.d.ts]: My bad, sorry for this. I've fixed now, hopefully no more tabs ;)

**Comment by thorn0** [comment]
Why do we need interfaces for something as simple as `{[binding: string]: string}`? But even if we need, why are these two different interfaces? They describe exactly the same thing: bindings.

**Comment by andy-ms** [comment]
@thorn0: You're right, sorry for merging before you could review. You could make a PR just for cleaning things up if you want.

**Comment by christophercr** [comment]
Well, indeed it's the same thing... I just kept them apart to make things
clear... but it could be merged in one single interface as you suggest...

**Pull request merged**